src/options/theme.rs

```diff
@@ -98,15 +100,91 @@ fn get_is_light_mode_and_syntax_theme_name(
     }
 }
 
+fn get_is_light(opt: &cli::Opt) -> bool {
+    get_is_light_opt(opt)
+        .or_else(|| should_detect_dark_light(opt).then(detect_light_mode))
+        .unwrap_or_default()
+}
+
+fn get_is_light_opt(opt: &cli::Opt) -> Option<bool> {
+    if opt.light {
+        Some(true)
+    } else if opt.dark {
+        Some(false)
+    } else {
+        None
+    }
+}
+
+fn should_detect_dark_light(opt: &cli::Opt) -> bool {
+    match opt.detect_dark_light {
+        // Querying the terminal for its colors requires "exclusive" access
+        // to the terminal since we read/write from the terminal and enable/disable raw mode.
+        // This causes race conditions with pagers such as less when they are attached to the
+        // same terminal as us.
+        //
+        // This is just an issue when the user manually pipes the output to less, e.g. `git diff | delta | less`.
+        // If we're the ones to start less, there's no race condition since less is started *after* we're done here.
```

Comment:
Would you mind moving this very helpful comment into the `--help` output? It answers the inevitable question "why wouldn't I want `always`?". I'm happy for the `--help` output to be long -- it's essentially a man page, and we have abbreviated output via the short form `-h`.

Reply:
Sure thing! I did a tiny bit of rewriting (mainly replaced us/we with delta).

> we have abbreviated output via the short form `-h`.

TIL 👀